WalletWasabi/Helpers/Constants.cs

```diff
@@ -103,9 +103,12 @@ public static class Constants
 		MinProtocolCapabilities = new ProtocolCapabilities { SupportGetBlock = true, SupportWitness = true }
 	};
 
-	public static readonly ExtPubKey FallBackCoordinatorExtPubKey = NBitcoinHelpers.BetterParseExtPubKey("xpub6BgAZqHhxw6pgEi2F38w5RBqctqCEoVWqcMdrn1epQZceKHtn8f8zHBduM3fwYQEKEGUf4efD6qRPc9wvDF4neoc6JjDbHNiaHbs3we5qL3");
-	public static readonly ExtPubKey WabiSabiFallBackCoordinatorExtPubKey = NBitcoinHelpers.BetterParseExtPubKey("xpub6C13JhXzjAhVRgeTcRSWqKEPe1vHi3Tmh2K9PN1cZaZFVjjSaj76y5NNyqYjc2bugj64LVDFYu8NZWtJsXNYKFb9J94nehLAPAKqKiXcebC");
-	public static readonly PubKey WasabiPubKey = new("02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608");
+	public static readonly Lazy<ExtPubKey> WabiSabiFallBackCoordinatorExtPubKey = new(
+		valueFactory: () => NBitcoinHelpers.BetterParseExtPubKey("xpub6C13JhXzjAhVRgeTcRSWqKEPe1vHi3Tmh2K9PN1cZaZFVjjSaj76y5NNyqYjc2bugj64LVDFYu8NZWtJsXNYKFb9J94nehLAPAKqKiXcebC"),
+		isThreadSafe: true);
+	public static readonly Lazy<PubKey> WasabiPubKey = new(
```

Comment:
I think this doesn't worth it because the cost/benefit is not clear, after all this is a very cheap operation. In fact, even if the operation were a bit more costly, like the one in L107, it would still be an extremely cheap operation and I wouldn't do it lazy.

Reply:
My comment is in a very specific line so, lets focus on that one first. Look, you have to be measuring things in the wrong way because doing a simple test you can see that the numbers are completely different and made sense:

I can create, using and interactive shell, 1 million public keys in 9.4 seconds, what means that it takes 0.0094 ms to create/parse one public key. Take into account that in this example I am using the slowest possible way to do it so, this is not fast only under normal circumstances but also under the worst possible circumstances too. How faster this is in a real program built in release mode? I think it should be much much much faster.  
 
```
> #r "nuget:nbitcoin";;
[Loading /home/lontivero/.packagemanagement/nuget/Projects/16321--6cb11e35-8130-4cfd-b2fa-f89ae3666109/Project.fsproj.fsx]
module FSI_0002.Project.fsproj

> open NBitcoin;;
> let hex = "02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608";;
val hex: string =
  "02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608"

> #time
- let pubkeys = seq {0..1_000_000} |> Seq.map (fun _ -> new PubKey(hex)) |> Seq.toArray
- #time
- ;;

--> Timing now on

Real: 00:00:09.426, CPU: 00:00:09.560, GC gen0: 0, gen1: 0, gen2: 0
val pubkeys: PubKey array =
  [|02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608;
    02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608;
    02c8ab8eea76c83788e246a1baee10c04a134ec11be6553946f6ae65e47ae9a608;
```

Reply:
No. Even if you your numbers were right this discussion wouldn't worth the effort. But the thing is that parsing a public key takes no time and then making it lazy makes no sense. 

Now, if the discussion is about initialization time then the solution is not making these line lazy but putting constants in the `Constants.cs` file and create/parse the public keys were are better, publishing the assembly as readytorun or who known but in anyway 80ms doesn't worth it.